app/src/main/java/openfoodfacts/github/scrachx/openfood/repositories/ProductRepository.java

```diff
@@ -308,6 +338,61 @@ public void saveCategories(List<Category> categories) {
         }
     }
 
+    /**
+     * Delete rows from Ingredient, IngredientName and IngredientsRelation
+     * set the autoincrement to 0
+     */
+    @Override
+    public void deleteIngredientCascade(){
+        ingredientDao.deleteAll();
+        ingredientNameDao.deleteAll();
+        ingredientsRelationDao.deleteAll();
+        DaoSession daoSession = OFFApplication.getInstance().getDaoSession();
+        daoSession.getDatabase().execSQL("update sqlite_sequence set seq=0 where name in ('" + ingredientDao.getTablename() + "', '" + ingredientNameDao.getTablename() + "', '" + ingredientsRelationDao.getTablename() + "')");
+    }
+
+    /**
+     * Ingredients saving to local database
+     * <p>
+     * Ingredient and IngredientName has One-To-Many relationship, therefore we need to save them separately.
+     */
+    @Override
+    public void saveIngredients(List<Ingredient> ingredients) {
+        db.beginTransaction();
+        try {
+            for (Ingredient ingredient : ingredients) {
+                ingredientDao.insertOrReplace(ingredient);
+                for (IngredientName ingredientName : ingredient.getNames()) {
+                    ingredientNameDao.insertOrReplace(ingredientName);
+                }
+                for (IngredientsRelation ingredientsRelation : ingredient.getParents()) {
+                    ingredientsRelationDao.insertOrReplace(ingredientsRelation);
+                }
+                for (IngredientsRelation ingredientsRelation : ingredient.getChildren()) {
+                    ingredientsRelationDao.insertOrReplace(ingredientsRelation);
+                }
+            }
+
+            db.setTransactionSuccessful();
+        } catch (Exception e) {
+            e.printStackTrace();
+        } finally {
+            db.endTransaction();
+        }
+    }
+
+    /**
+     * Ingredient saving to local database
+     */
+    @Override
+    public void saveIngredient(Ingredient ingredient) {
```

Comment:
usefull ?
Not used I guess. Could it be removed or will be used ?

Reply:
Same comment.